File: 2390-removing-stars-from-a-string/2390-removing-stars-from-a-string.cpp
class Solution {
public:
    string removeStars(string s) {
        string res = "";
        for (char& c : s) {
            if (c != '*') {
                res.push_back(c);
            } else {
                res.pop_back();
            }
        }
        return res;
    }
};

File: 2390-removing-stars-from-a-string/2390-removing-stars-from-a-string.java
class Solution {
    public String removeStars(String s) {
        StringBuilder builder = new StringBuilder();
        for (char c : s.toCharArray()) {
            if (c == '*') {
                builder.setLength(builder.length() - 1);
            } else {
                builder.append(c);
            }
        }
        return builder.toString();
    }
}